src/evict/evict_page.c

```diff
@@ -663,7 +663,8 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
     /* Make sure that both conditions above are not true at the same time. */
     WT_ASSERT(session, !use_snapshot_for_app_thread || !is_eviction_thread);
 
-    if (!conn->txn_global.checkpoint_running && !WT_IS_HS(btree->dhandle) &&
+    checkpoint_running = conn->txn_global.checkpoint_running;
```

Comment:
I can't remember but we don't need to lock to access this?

Reply:
@hbokhari Can you explain why there is no barrier?

Reply:
Generally I think this is fine. There is only one place we set this flag to true, which is at the start of the checkpoint.

Reply:
The other threads are just reading the value.

Reply:
Yeah but the order of these reads appears to be important, I think gcc can re-order non volatile reads but as this is a volatile it should be okay?

Reply:
I think the order only matters for the write thread and read thread? It doesn't matter between reads. Within this thread, the read cannot be reordered.

Reply:
I believe the variable `conn->txn_global.checkpoint_running` is only being set in the checkpoint thread and only get tread in other threads. There is no need for lock in that scenario.